waiter/src/waiter/core.clj

```diff
@@ -120,35 +120,37 @@
   "Creates the handler for processing http requests."
   [waiter-request?-fn {:keys [cors-preflight-handler-fn process-request-fn] :as handlers}]
   (fn http-handler [{:keys [uri] :as request}]
-    (cond
-      (cors/preflight-request? request)
-      (do
-        (counters/inc! (metrics/waiter-counter "requests" "cors-preflight"))
-        (cors-preflight-handler-fn request))
+    (let [request (utils/mark-request-time request :received)]
+      (cond
+        (cors/preflight-request? request)
+        (do
+          (counters/inc! (metrics/waiter-counter "requests" "cors-preflight"))
+          (cors-preflight-handler-fn request))
 
-      (not (waiter-request?-fn request))
-      (do
-        (counters/inc! (metrics/waiter-counter "requests" "service-request"))
-        (process-request-fn request))
+        (not (waiter-request?-fn request))
+        (do
+          (counters/inc! (metrics/waiter-counter "requests" "service-request"))
+          (process-request-fn request))
 
-      :else
-      (let [{:keys [handler route-params]} (routes-mapper request)
-            request (assoc request :route-params (or route-params {}))
-            handler-fn (get handlers handler process-request-fn)]
-        (when (and (not= handler :process-request-fn) (= handler-fn process-request-fn))
-          (log/warn "using default handler as no mapping found for" handler "at uri" uri))
-        (when handler
-          (counters/inc! (metrics/waiter-counter "requests" (name handler))))
-        (handler-fn request)))))
+        :else
+        (let [{:keys [handler route-params]} (routes-mapper request)
+              request (assoc request :route-params (or route-params {}))
+              handler-fn (get handlers handler process-request-fn)]
+          (when (and (not= handler :process-request-fn) (= handler-fn process-request-fn))
+            (log/warn "using default handler as no mapping found for" handler "at uri" uri))
+          (when handler
+            (counters/inc! (metrics/waiter-counter "requests" (name handler))))
+          (handler-fn request))))))
 
 (defn websocket-handler-factory
   "Creates the handler for processing websocket requests.
    Websockets are currently used for inter-router metrics syncing."
   [{:keys [default-websocket-handler-fn router-metrics-handler-fn]}]
   (fn websocket-handler [{:keys [uri] :as request}]
-    (case uri
-      "/waiter-router-metrics" (router-metrics-handler-fn request)
-      (default-websocket-handler-fn request))))
+    (let [request (utils/mark-request-time request :received)]
```

Comment:
We now need to be able to exclude data introduced via these calls whenever we log  the request.

Reply:
Do we need to exclude the data?

Reply:
I think so, we may not need to view everything in here in our standard logs if we ever log the request.

Reply:
OK we can do an exclude pass when we're happy with everything that's in the log.